hfile.c

```diff
@@ -506,12 +518,13 @@ static off_t fd_seek(hFILE *fpv, off_t offset, int whence)
 
 static int fd_flush(hFILE *fpv)
 {
-    hFILE_fd *fp = (hFILE_fd *) fpv;
-    int ret;
+    int ret = 0;
     do {
 #ifdef HAVE_FDATASYNC
+        hFILE_fd *fp = (hFILE_fd *) fpv;
         ret = fdatasync(fp->fd);
-#else
+#elif defined(HAVE_FSYNC)
+        hFILE_fd *fp = (hFILE_fd *) fpv;
         ret = fsync(fp->fd);
```

Comment:
I don't get what you mean.  If we define ```HAVE_FLUSHFILEBUFFERS``` then we need our own custom autoconf code to generate that define, instead of using the easy existing macros for ```HAVE_```func style checking, which sounds more invasive rather than less.

Could you give an example of what you mean plase?